code/modules/mob/living/simple_animal/hostile/mining/hivelord.dm

```diff
@@ -0,0 +1,319 @@
+	icon = 'icons/mob/lavaland/lavaland_monsters.dmi'
+	icon_state = "Hivelord"
+	icon_living = "Hivelord"
+	icon_aggro = "Hivelord_alert"
+	icon_dead = "Hivelord_dead"
+	icon_gib = "syndicate_gib"
+	mouse_opacity = 2
+	move_to_delay = 14
+	ranged = 1
+	vision_range = 5
+	aggro_vision_range = 9
+	idle_vision_range = 5
+	speed = 3
+	maxHealth = 75
+	health = 75
+	harm_intent_damage = 5
+	melee_damage_lower = 0
+	melee_damage_upper = 0
+	attacktext = "lashes out at"
+	speak_emote = list("telepathically cries")
+	attack_sound = 'sound/weapons/pierce.ogg'
+	throw_message = "falls right through the strange body of the"
+	ranged_cooldown = 0
+	ranged_cooldown_time = 20
+	environment_smash = 0
+	retreat_distance = 3
+	minimum_distance = 3
+	pass_flags = PASSTABLE
+	loot = list(/obj/item/organ/internal/hivelord_core)
+	var/brood_type = /mob/living/simple_animal/hostile/asteroid/hivelordbrood
+
+/mob/living/simple_animal/hostile/asteroid/hivelord/OpenFire(the_target)
+	if(world.time >= ranged_cooldown)
+		var/mob/living/simple_animal/hostile/asteroid/hivelordbrood/A = new brood_type(src.loc)
+		A.admin_spawned = admin_spawned
+		A.GiveTarget(target)
+		A.friends = friends
+		A.faction = faction.Copy()
+		ranged_cooldown = world.time + ranged_cooldown_time
+
+/mob/living/simple_animal/hostile/asteroid/hivelord/AttackingTarget()
+	OpenFire()
+
+/mob/living/simple_animal/hostile/asteroid/hivelord/death(gibbed)
+	mouse_opacity = 1
+	..(gibbed)
+
+/obj/item/organ/internal/hivelord_core
+	name = "hivelord remains"
+	desc = "All that remains of a hivelord, it seems to be what allows it to break pieces of itself off without being hurt... its healing properties will soon become inert if not used quickly."
+	icon_state = "roro core 2"
+	flags = NOBLUDGEON
+	slot = "hivecore"
+	force = 0
+	actions_types = list(/datum/action/item_action/organ_action/use)
+	var/inert = 0
+	var/preserved = 0
+
+/obj/item/organ/internal/hivelord_core/New()
+	..()
+	addtimer(src, "inert_check", 2400)
+
+/obj/item/organ/internal/hivelord_core/proc/inert_check()
+	if(!owner && !preserved)
+		go_inert()
+	else
+		preserved(implanted = 1)
+
+/obj/item/organ/internal/hivelord_core/proc/preserved(implanted = 0)
+	inert = FALSE
+	preserved = TRUE
+	update_icon()
+
+	if(implanted)
+		feedback_add_details("hivelord_core", "[type]|implanted")
+	else
+		feedback_add_details("hivelord_core", "[type]|stabilizer")
+
+
+/obj/item/organ/internal/hivelord_core/proc/go_inert()
+	inert = TRUE
+	desc = "The remains of a hivelord that have become useless, having been left alone too long after being harvested."
+	feedback_add_details("hivelord_core", "[src.type]|inert")
+	update_icon()
+
+/obj/item/organ/internal/hivelord_core/ui_action_click()
+	owner.revive()
+	qdel(src)
+
+/obj/item/organ/internal/hivelord_core/on_life()
+	..()
+	if(owner.health < config.health_threshold_crit)
+		ui_action_click()
```

Comment:
Ah, the hivelord core is now an extra life, instead of a permanent auto-regen, now - noted.